Sources/SourceKitBazelBSP/RequestHandlers/BuildTargets/BazelQueryParser.swift

```diff
@@ -0,0 +1,166 @@
+                guard let srcsEntryElement = srcsEntry as? XMLElement else {
+                    continue
+                }
+                let srcValue = srcsEntryElement.attribute(forName: "value")?.stringValue ?? ""
+                // FIXME
+                if !srcValue.starts(with: "//") {
+                    continue
+                }
+                let src = srcValue.replacingOccurrences(of: ":", with: "/")
+                let srcUri = try URI(string: "file://" + rootUri + "/" + src.dropFirst(2))
+                targetSrcs.append(srcUri)
+            }
+        }
+
+        var targetDeps: [BuildTargetIdentifier] = []
+        for child in (childElement.children ?? []) {
+            if child.name != "list" {
+                continue
+            }
+            guard let childElement = child as? XMLElement else {
+                continue
+            }
+            let name = childElement.attribute(forName: "name")?.stringValue ?? ""
+            guard name == "deps" else {
+                continue
+            }
+            for depsEntry in (childElement.children ?? []) {
+                if depsEntry.name != "label" {
+                    continue
+                }
+                guard let depsEntryElement = depsEntry as? XMLElement else {
+                    continue
+                }
+                let depValue = depsEntryElement.attribute(forName: "value")?.stringValue ?? ""
+                // FIXME
+                if !depValue.starts(with: "//") {
+                    continue
+                }
+                let depFullPath = rootUri + "/" + depValue.dropFirst(2)
+                let depUri = bazelTargetToURI(depFullPath)
+                targetDeps.append(BuildTargetIdentifier(uri: try URI(string: depUri)))
+            }
+        }
+
+        var tags: [BuildTargetTag] = [.library]
+        var capabilities = BuildTargetCapabilities(
+            canCompile: true,
+            canTest: false,
+            canRun: false,
+            canDebug: false
+        )
+        // FIXME: Not the way to do this
+        if bazelTarget.hasSuffix("TestsLib") {
+            capabilities.canTest = true
+            tags.append(.test)
+        }
+        return (
+            BuildTarget(
+                id: BuildTargetIdentifier(uri: uri),
+                displayName: bazelTarget,
+                baseDirectory: try URI(string: basePath),
+                tags: tags,
+                capabilities: capabilities,
+                languageIds: isSwift ? [.swift] : [.objective_c],
+                dependencies: targetDeps,
+                dataKind: .sourceKit,
+                data: SourceKitBuildTarget(
+                    toolchain: try URI(string: "file://" + toolchainPath)
+                ).encodeToLSPAny()
+            ), targetSrcs
+        )
+    }
+
+    static func bazelTargetToURI(_ bazelTarget: String) -> String {
+        return "file://\(bazelTarget.replacingOccurrences(of: ":", with: "___"))"
```

Comment:
you probably don't need to replace colon here, since for file scheme URI, `:` colon is actually valid character